### PR TITLE
feat: add Camie ONNX tagger

### LINKED ISSUE #89: Add CamieTagger ONNX adapter support
## Summary

Implement CamieTagger support as the second part of ADR 0004. Camie should be treated as a large ONNX tagger, not as a rating-only model.

## Scope

- Add a `CamieTagger` adapter using the generalized ONNX tagger loader.
- Resolve Camie ONNX model and JSON metadata files from Hugging Face.
- Parse Camie metadata to recover labels and categories.
- Apply the required sigmoid post-processing to model logits.
- Return only high-value categories initially:
  - `rating`
  - `general`
  - `character`
- Return rating output through `UnifiedAnnotationResult.ratings` with `source_scheme = "danbooru4"`.
- Do not mix rating labels into `tags`.
- Document that Camie model files are GPL-3.0 and are not bundled with the library.

## Acceptance Criteria

- `CamieTagger` can load ONNX model and JSON metadata via the generalized loader.
- `CamieTagger` returns content tags from `general` and `character`.
- `CamieTagger` returns top-1 rating in `ratings`.
- `artist`, `copyright`, `meta`, and `year` are not returned by default.
- GPL-3.0 model license is mentioned in docs/model specs.
- Unit tests cover metadata parsing, sigmoid formatting, category filtering, and rating extraction.
- Runtime validation is available behind the existing heavy/download marker.

## Out of Scope

- LoRAIro canonical rating mapping.
- Bundling Camie model weights or metadata.
- Copying Camie official inference code or dghs-imgutils implementation code.
- Returning lower-confidence Camie categories by default.

## References

- ADR 0004: `docs/decisions/0004-onnx-tagger-loader-and-model-selection.md`
- Model specs: `docs/model-specs/rating-model-candidates.md`
- Camie model card: https://huggingface.co/Camais03/camie-tagger
- DeepGHS ONNX export: https://huggingface.co/deepghs/camie_tagger_onnx
- Reference docs: https://dghs-imgutils.deepghs.org/main/api_doc/tagging/camie.html

### PR DESCRIPTION
**Pull request by NEXTAltair**
## Summary
- add CamieTagger as a large ONNX tagger using generalized model/metadata file loading
- parse Camie JSON metadata and expose rating/general/character only
- apply Camie sigmoid post-processing and normalize rating_* labels to danbooru4 labels
- document/register camie_tagger_initial without bundling GPL-3.0 model files

## Stacked PR
- Depends on #90 / issue #88.
- Base branch is feat/issue-88-onnx-loader until #90 is merged.

## Tests
- uv run ruff check src/image_annotator_lib/model_class/tagger_onnx.py src/image_annotator_lib/core/model_config.py tests/unit/model_class/test_camie_tagger.py tests/runtime_validation/test_real_model_runtime.py
- uv run ruff format --check src/image_annotator_lib/model_class/tagger_onnx.py src/image_annotator_lib/core/model_config.py tests/unit/model_class/test_camie_tagger.py tests/runtime_validation/test_real_model_runtime.py
- uv run pytest tests/unit/model_class/test_camie_tagger.py tests/unit/core/test_onnx_loader.py tests/unit/model_class/test_tagger_onnx.py tests/unit/fast/test_list_annotator_info.py
- Added but not run: tests/runtime_validation/test_real_model_runtime.py::test_real_camie_tagger_runtime -m downloads_and_runs_model (heavy ~856MB model download)

Closes #89

### CODE REVIEW

**Review by NEXTAltair — COMMENTED** [review]
## コードレビュー (CamieTagger)

`ONNXBaseAnnotator` / 既存 `WDTagger` / `Z3D_E621Tagger` と照合。correctness を壊すバグはなし。指摘事項のうち主要なものは commit `940d692` で対応済み。

### 対応済み (`940d692`)

- **カテゴリ情報を削らない**: `CamieTagger` が general/character/rating のみ保持し copyright/artist/meta/year を破棄していたのを、metadata の全カテゴリを動的に `_category_attr_map` へ取り込む方式へ変更。rating 以外の全カテゴリのタグが出力され、per-category raw スコアは `raw_output["category_scores"]` に残る。どのカテゴリをどう使うかは consumer (LoRAIro) 側で判断する方針。
- **`_format_predictions_single` の重複解消**: 基底クラスのメソッドを約40行コピーし sigmoid だけ差し込む実装だった (`WDTagger`/`Z3D` は誰も override せず基底に乗っている)。基底に `_to_probabilities` hook を追加し、`CamieTagger` は logit→sigmoid 変換のみ override。基底の今後の変更が伝播するようになり、`raw_output` dict の `predictions` キー欠落の不整合も解消。確率を直接出す WD/Z3D は既定の恒等変換で挙動不変。
- **`_preprocess_images` の冗長条件**: `image.mode in {"RGBA","P"} or image.mode != "RGB"` は後者が前者を包含。`mode != "RGB"` へ単純化。

### 未対応 (minor / 確認事項)

- **RGBA の扱いが基底と非一貫**: 基底 `_preprocess_images` は RGBA を白背景へ alpha 合成するが、Camie 版は `.convert("RGB")` で alpha を単純破棄。透過 PNG の扱いが他 ONNX tagger と変わる。意図的か要確認。
- **NCHW ハードコード**: `_analyze_model_input_format` が `is_nchw_expected` を判定済みだが、Camie 版は無条件に `transpose(2,0,1)`。Camie ONNX は NCHW 固定なので動くが `is_nchw_expected` 参照の方が頑健。
- **`idx_to_tag` の連続性無検証**: `[idx_to_tag[str(i)] for i in range(len(idx_to_tag))]` はキー連続前提。欠番時に素の `KeyError`。
- **`model_initial.onnx` のハードコード**: `LocalMLModelConfig.model_variant` があるのに未使用。将来 Camie full モデル追加時に `CamieTagger` を再利用しづらい。
- **sigmoid の数値安定性**: 大きい負 logit で `np.exp` オーバーフロー警告 (実値は正しい)。

### プロセス

- PR の Tests 欄に **mypy がない**。なお基底 `onnx.py` の `_format_predictions_single` には PR #91 以前からの既存 mypy エラー2件 (`predictions: ndarray | None` の絞り込み不足) がある。本 PR の新規コード由来ではないため未修正。別途対応推奨。
- `test_real_camie_tagger_runtime` (`downloads_and_runs_model`) は未実行。`model_initial.onnx` の単一出力前提はこの heavy test でしか検証できないため、マージ前後で一度実行を推奨。

### 検証

CI-equivalent suite **699 passed**、変更ファイルの ruff check / format クリーン。`test_camie_tagger.py` は全カテゴリ保持の挙動へ更新済み。
